jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -717,9 +713,16 @@ protected void writeHttpContent(Request request, Response response, Callback cal
         {
             ByteBuffer buffer = content.getByteBuffer(); // this buffer is going to be consumed by response.write()
             if (buffer != null)
+            {
                 response.write(true, buffer, callback);
+            }
             else
-                new ContentWriterIteratingCallback(content, response, callback).iterate();
+            {
+                ByteBufferPool bufferPool = request.getComponents().getByteBufferPool();
+                int outputBufferSize = request.getConnectionMetaData().getHttpConfiguration().getOutputBufferSize();
+                boolean useOutputDirectByteBuffers = request.getConnectionMetaData().getHttpConfiguration().isUseOutputDirectByteBuffers();
+                IOResources.copy(content.getResource(), bufferPool, outputBufferSize, useOutputDirectByteBuffers, response, callback);
+            }
```

Comment:
I think the following is actually more readable:
```suggestion
                IOResources.copy(
                    content.getResource(), 
                    request.getComponents().getByteBufferPool(), 
                    request.getConnectionMetaData().getHttpConfiguration().getOutputBufferSize(), 
                 
   request.getConnectionMetaData().getHttpConfiguration().isUseOutputDirectByteBuffers(),
                   response,
                   callback);
            }
```

Reply:
Done.